lib/edgehog_device_forwarder/monitor.ex

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.Monitor do
+  use GenServer
+
+  @spec monitor(pid, (-> any)) :: :ok
+  def monitor(pid, on_close) when is_function(on_close, 0) do
+    GenServer.cast(__MODULE__, {:monitor, pid, on_close})
+  end
```

Comment:
`Monitor` (both ad module name and function name) is a little generic and doesn't convey what this actually does. I'd call this something like `TerminationHandler` and `clean_up_with` (or propose some other names)

Reply:
I propose `TerminationCallbacks` and `add`